Fix model so that it works with Xyce and ngspice

Xyce gets confused when a node name in an expression contains + or -;
rename + -> INP, - -> INN, V+ - VCC, V- -> VEE.

ngspice treats $ as an inline comment; rename $N_foo to N_foo.

ngspice has does not understand the switch model syntax used for
OPA695_model_S1; hardcode the switch resistances.

ngspice expects the PNP transistor area to be specified as area=value
while other simulators expect only the number.  Add an _AREA define
which evaluates to area=.

--- models_source/ti/op_amp/OPA695_PSPICE_MODEL.txt	2018-08-29 02:20:26.379136632 +0200
+++ models/ti/op_amp/OPA695.lib	2018-08-29 04:11:29.479772707 +0200
@@ -29,11 +29,17 @@
 *     input voltage range, input voltage offset and output voltage range.
 ********************************************************************************

-.SUBCKT OPA695 + -  V+ V- Out DisA
-X1 + - DisA Out V+ V- OPA695_model
+.SUBCKT OPA695 INP INN VCC VEE Out DisA
+X1 INP INN DisA Out VCC VEE OPA695_model
 .ENDS

-.SUBCKT OPA695_model + - Dis OUT V+ V-
+#ifdef NGSPICE
+#define _AREA area=
+#else
+#define _AREA
+#endif
+
+.SUBCKT OPA695_model INP INN Dis OUT VCC VEE

 .PARAM x1 = {1/8}
 .PARAM x2 = {x1*2}
@@ -44,85 +50,90 @@
 .PARAM x60 = {x1*60}

 C_C1         Vmid Dis  2p
-X_S1    $N_0001 0 $N_0002 $N_0003 OPA695_model_S1
+#ifdef NGSPICE
+R_S1A      N_0001 0 1e9
+R_S1B      N_0002 N_0003 0.001
+#else
+X_S1    N_0001 0 N_0002 N_0003 OPA695_model_S1
+#endif
 R_R11         Vmid Dis  100MEG
-R_R8         Vmid $N_0004  10k
-Q_Q1         $N_0006 $N_0005 $N_0007 PNP8 {x1}
-Q_Q4         $N_0008 $N_0008 $N_0005 NPN8 {x1}
-Q_Q5         $N_0009 $N_0005 $N_0010 PNP8 {x2}
-Q_Q12         $N_0011 $N_0005 $N_0012 PNP8 {x2}
-Q_Q9         $N_0013 $N_0005 $N_0014 PNP8 {x2}
-Q_Q7         $N_0015 $N_0015 $N_0016 NPN8 {x2}
-Q_Q6         $N_0015 $N_0017 $N_0009 PNP8 {x16}
-Q_Q14         $N_0011 $N_0017 $N_0018 NPN8 {x1}
-Q_Q16         $N_0017 $N_0018 $N_0019 NPN8 {x1}
-Q_Q15         $N_0018 $N_0018 $N_0020 NPN8 {x1}
-Q_Q10         $N_0011 $N_0011 $N_0021 NPN8 {x16}
-Q_Q8         $N_0013 $N_0015 $N_0022 NPN8 {x2}
-D_D3         $N_0023 $N_0021 DX 1
-D_D1         $N_0021 $N_0024 DX 1
-D_D2         $N_0024 $N_0025 DX 1
-Q_Q18         $N_0026 $N_0026 $N_0021 PNP8 {x16}
-Q_Q17         $N_0026 $N_0015 $N_0027 NPN8 {x2}
-R_R24         V- $N_0028  250
-X_F1    $N_0002 $N_0029 $N_0006 V- OPA695_model_F1
-Q_Q2         V- $N_0006 $N_0005 PNP8 {x1}
-R_R18         V- $N_0015  50k
-R_R20         V- $N_0022  500
-R_R19         V- $N_0016  500
-R_R21         V- $N_0019  2k
-R_R22         V- $N_0020  2k
-R_R23         V- $N_0027  500
-D_D5         $N_0028 V- DX 1
-Q_Q20         $N_0030 $N_0030 $N_0028 NPN8 {x3}
-Q_Q19         $N_0030 $N_0026 $N_0025 PNP8 {x16}
-Q_Q25         $N_0031 $N_0030 $N_0032 NPN8 {x12}
-Q_Q24         $N_0034 $N_0033 $N_0035 PNP8 {x12}
-R_R10         V+ V-  123.609k
-E_E1         $N_0004 V- V+ V- 0.5
-E_ABM21         $N_0001 0 VALUE { ( V(Dis)
-+ / (V(V+)
+R_R8         Vmid N_0004  10k
+Q_Q1         N_0006 N_0005 N_0007 PNP8 _AREA{x1}
+Q_Q4         N_0008 N_0008 N_0005 NPN8 _AREA{x1}
+Q_Q5         N_0009 N_0005 N_0010 PNP8 _AREA{x2}
+Q_Q12         N_0011 N_0005 N_0012 PNP8 _AREA{x2}
+Q_Q9         N_0013 N_0005 N_0014 PNP8 _AREA{x2}
+Q_Q7         N_0015 N_0015 N_0016 NPN8 _AREA{x2}
+Q_Q6         N_0015 N_0017 N_0009 PNP8 _AREA{x16}
+Q_Q14         N_0011 N_0017 N_0018 NPN8 _AREA{x1}
+Q_Q16         N_0017 N_0018 N_0019 NPN8 _AREA{x1}
+Q_Q15         N_0018 N_0018 N_0020 NPN8 _AREA{x1}
+Q_Q10         N_0011 N_0011 N_0021 NPN8 _AREA{x16}
+Q_Q8         N_0013 N_0015 N_0022 NPN8 _AREA{x2}
+D_D3         N_0023 N_0021 DX 1
+D_D1         N_0021 N_0024 DX 1
+D_D2         N_0024 N_0025 DX 1
+Q_Q18         N_0026 N_0026 N_0021 PNP8 _AREA{x16}
+Q_Q17         N_0026 N_0015 N_0027 NPN8 _AREA{x2}
+R_R24         VEE N_0028  250
+X_F1    N_0002 N_0029 N_0006 VEE OPA695_model_F1
+Q_Q2         VEE N_0006 N_0005 PNP8 _AREA{x1}
+R_R18         VEE N_0015  50k
+R_R20         VEE N_0022  500
+R_R19         VEE N_0016  500
+R_R21         VEE N_0019  2k
+R_R22         VEE N_0020  2k
+R_R23         VEE N_0027  500
+D_D5         N_0028 VEE DX 1
+Q_Q20         N_0030 N_0030 N_0028 NPN8 _AREA{x3}
+Q_Q19         N_0030 N_0026 N_0025 PNP8 _AREA{x16}
+Q_Q25         N_0031 N_0030 N_0032 NPN8 _AREA{x12}
+Q_Q24         N_0034 N_0033 N_0035 PNP8 _AREA{x12}
+R_R10         VCC VEE  123.609k
+E_E1         N_0004 VEE VCC VEE 0.5
+E_ABM21         N_0001 0 VALUE={ ( V(Dis)
++ / (V(VCC)
 + + 1e-6) ) * 5.0  }
-R_R9         Dis V+  41k
-R_R12         $N_0008 V+  8k
-R_R14         $N_0010 V+  500
-R_R15         $N_0014 V+  500
-R_R16         $N_0012 V+  500
-R_R17         $N_0006 V+  72k
-R_R13         $N_0007 V+  1987
-Q_Q21         $N_0036 $N_0030 $N_0037 NPN8 {x5}
-R_R7         $N_0029 Vmid  10k
-R_R25         V- $N_0037  150
-D_D7         V+ $N_0038 DX 1
-R_R26         $N_0038 V+  250
-Q_Q22         $N_0033 $N_0033 $N_0038 PNP8 {x3}
-R_R27         $N_0039 V+  150
-Q_Q23         $N_0036 $N_0033 $N_0039 PNP8 {x5}
-D_D8         $N_0039 V+ DX 1
-Q_Q13         $N_0017 $N_0006 $N_0013 PNP8 {x1}
-Q_Q30         $N_0041 $N_0040 $N_0036 NPN8 {x1}
-Q_Q31         $N_0042 $N_0040 $N_0036 PNP8 {x1}
-V_V1         $N_0003 Vmid 2.04V
-L_L1         + $N_0021  1.3nH
-Q_Q11         $N_0033 $N_0011 $N_0025 NPN8 {x16}
-D_D4         $N_0025 $N_0023 DX 1
-L_L3         $N_0040 OUT  1.3nH
-L_L2         $N_0025 $N_0043  1.3nH
-C_C2         $N_0036 V+  0.2p
-R_R31         $N_0043 -  65
-Q_Q32         $N_0044 $N_0044 $N_0041 PNP8 {x60}
-Q_Q33         $N_0040 $N_0044 $N_0041 PNP8 {x60}
-Q_Q28         $N_0045 $N_0045 $N_0042 NPN8 {x60}
-Q_Q29         $N_0040 $N_0045 $N_0042 NPN8 {x60}
-Q_Q35         $N_0045 $N_0031 $N_0040 PNP8 {x60}
-Q_Q34         $N_0044 $N_0034 $N_0040 NPN8 {x60}
-Q_Q27         $N_0042 $N_0036 $N_0034 PNP8 {x60}
-Q_Q26         $N_0041 $N_0036 $N_0031 NPN8 {x60}
-R_R28         $N_0035 V+  62.5
-D_D6         $N_0037 V- DX 1
-R_R29         V- $N_0032  62.5
-V_V3         V- $N_0042 0.2V
-V_V2         $N_0041 V+ 0.2V
+R_R9         Dis VCC  41k
+R_R12         N_0008 VCC  8k
+R_R14         N_0010 VCC  500
+R_R15         N_0014 VCC  500
+R_R16         N_0012 VCC  500
+R_R17         N_0006 VCC  72k
+R_R13         N_0007 VCC  1987
+Q_Q21         N_0036 N_0030 N_0037 NPN8 _AREA{x5}
+R_R7         N_0029 Vmid  10k
+R_R25         VEE N_0037  150
+D_D7         VCC N_0038 DX 1
+R_R26         N_0038 VCC  250
+Q_Q22         N_0033 N_0033 N_0038 PNP8 _AREA{x3}
+R_R27         N_0039 VCC  150
+Q_Q23         N_0036 N_0033 N_0039 PNP8 _AREA{x5}
+D_D8         N_0039 VCC DX 1
+Q_Q13         N_0017 N_0006 N_0013 PNP8 _AREA{x1}
+Q_Q30         N_0041 N_0040 N_0036 NPN8 _AREA{x1}
+Q_Q31         N_0042 N_0040 N_0036 PNP8 _AREA{x1}
+V_V1         N_0003 Vmid 2.04V
+L_L1         INP N_0021  1.3nH
+Q_Q11         N_0033 N_0011 N_0025 NPN8 _AREA{x16}
+D_D4         N_0025 N_0023 DX 1
+L_L3         N_0040 OUT  1.3nH
+L_L2         N_0025 N_0043  1.3nH
+C_C2         N_0036 VCC  0.2p
+R_R31         N_0043 INN  65
+Q_Q32         N_0044 N_0044 N_0041 PNP8 _AREA{x60}
+Q_Q33         N_0040 N_0044 N_0041 PNP8 _AREA{x60}
+Q_Q28         N_0045 N_0045 N_0042 NPN8 _AREA{x60}
+Q_Q29         N_0040 N_0045 N_0042 NPN8 _AREA{x60}
+Q_Q35         N_0045 N_0031 N_0040 PNP8 _AREA{x60}
+Q_Q34         N_0044 N_0034 N_0040 NPN8 _AREA{x60}
+Q_Q27         N_0042 N_0036 N_0034 PNP8 _AREA{x60}
+Q_Q26         N_0041 N_0036 N_0031 NPN8 _AREA{x60}
+R_R28         N_0035 VCC  62.5
+D_D6         N_0037 VEE DX 1
+R_R29         VEE N_0032  62.5
+V_V3         VEE N_0042 0.2V
+V_V2         N_0041 VCC 0.2V

 .MODEL NPN8 NPN
 + IS = 7.604E-18 BF = 1.570E+02 NF = 1.000E+00 VAF= 7.871E+01
@@ -166,3 +177,5 @@
 F_F1         3 4 VF_F1 1
 VF_F1         1 2 0V
 .ends OPA695_model_F1
+
+#undef _AREA
